terraform/examples/offline.tf

```diff
@@ -0,0 +1,761 @@
+# This file is meant to define the 'offline' environment, which lives in the 'crash' trust zone.
+# See also https://github.com/zinfra/backend-issues/wiki/trust-zone-environments (TODO: move that page over to either the zinfra/backend-wiki's wiki, or to zinfra/backend-wiki's repository)
+
+# What is an Offline VPC?
+# The only host reachable from the outside is a bastion host. It has internet access, and provides SSH services. It can access all of the other hosts in the VPC via SSH only.
+
+# To deploy this file, you will need a user with the policies "AmazonEC2FullAccess", "IAMFullAccess", "AmazonS3FullAccess", and "AmazonDynamoDBFullAccess".
+# FUTUREWORK: drill down on the above.
+
+terraform {
+  required_version = ">= 0.12.0"
+
+  backend "s3" {
+    encrypt = true
+    region  = "eu-central-1"
+
+    # TODO: create IAM policy which only allows access to this bucket under
+    # envrionments/crash and to the -crash dynamodb table
+    bucket = "z-terraform-remote-state"
+
+    key = "environments/offline/bootstrap"
+
+    dynamodb_table = "z-terraform-state-lock-dynamo-lock-environment-offline"
+  }
```

Comment:
Remove the entire `backend "s3"` block (leaving only the version constraint) - it's not needed inside a module, as module users will define this, not the modules themselves